compiler/rustc_ast/src/ast.rs

```diff
@@ -3077,8 +3092,10 @@ mod size_asserts {
     static_assert_size!(AssocItemKind, 32);
     static_assert_size!(Attribute, 32);
     static_assert_size!(Block, 48);
-    static_assert_size!(Expr, 104);
-    static_assert_size!(ExprKind, 72);
+    #[cfg(not(bootstrap))]
```

Comment:
Not sure why this `cfg` is needed.
Aren't all these asserts behave identically with regards to bootstrap changes? Why is only `Expr(Kind)` needs to be configured?

Reply:
#94075 recently landed, changing the layout of some types, and shrinking some of them. The `Fn` size assertion in this file is also conditional, as are some in other files, e.g. [hir.rs](https://github.com/rust-lang/rust/blob/6153d3cbe6abc74fb37e4ebe48cc825484fd6bbf/compiler/rustc_hir/src/hir.rs#L3512-L3550).